File: backend/app/schemas/personal_transaction.py
from __future__ import annotations

from datetime import date, datetime
from decimal import Decimal

from ..models.enums import EnumPersonalTransactionType
from .base import BaseSchema


class PersonalTransactionRead(BaseSchema):
    id: str
    date: date
    category: str
    subcategory: str | None = None
    amount: Decimal
    type: EnumPersonalTransactionType
    payment_method: str
    description: str
    memo: str | None = None
    hash: str
    created_at: datetime
    updated_at: datetime
    organization_id: str | None = None

File: frontend/src/types/personal-transaction.ts
export type PersonalTransactionType = "income" | "expense";

export interface PersonalTransaction {
  id: string;
  date: Date;
  category: string;
  subcategory?: string;
  amount: number;
  type: PersonalTransactionType;
  payment_method: string;
  description: string;
  memo?: string;
  created_at: Date;
  updated_at: Date;
  organizationId?: string;
}

export interface PersonalTransactionFilters {
  organizationSlug?: string;
  category?: string;
  type?: PersonalTransactionType;
  dateFrom?: Date;
  dateTo?: Date;
  limit?: number;
  offset?: number;
}

export interface PersonalTransactionSummary {
  totalCount: number;
  totalIncome: number;
  totalExpense: number;
  categories: Array<{
    category: string;
    count: number;
    totalAmount: number;
  }>;
}
